ci/Jenkinsfile-kubernetes

```diff
@@ -49,32 +49,46 @@ def getBuildAndPublishStage(props, sparkMajorVersion) {
                         sh "sudo service docker start"
                         sh "sudo chmod 666 /var/run/docker.sock"
                         def sparklingVersion = getSparklingVersion(props, sparkMajorVersion)
+                        def sparkVersion = getSparkVersion(sparkMajorVersion)
                         sh "./gradlew dist -Pspark=$sparkMajorVersion"
                         def customEnv = [
-                                "SPARK_HOME=/home/jenkins/spark-${getSparkVersion(sparkMajorVersion)}-bin-hadoop2.7",
+                                "SPARK_HOME=/home/jenkins/spark-${sparkVersion}-bin-hadoop2.7",
                         ]
-                        withEnv(customEnv) {
-                            dir("./dist/build/zip/sparkling-water-${sparklingVersion}") {
-                                sh """
-                                    ./bin/build-kubernetes-images.sh scala
-                                    ./bin/build-kubernetes-images.sh python
-                                    ./bin/build-kubernetes-images.sh r
-                                   """
-                            }
-                        }
-
                         unstash "properties"
                         def repoId = readPropertiesFile("properties")["docker_registry_id"]
                         def repoUrl = "${repoId}.dkr.ecr.us-east-2.amazonaws.com"
                         docker.withRegistry("https://${repoUrl}", 'ecr:us-east-2:SW_FULL_AWS_CREDS') {
-                            sh """
-                            docker tag sparkling-water-r:${sparklingVersion} $repoUrl/sw_kubernetes_repo/sparkling-water:r-${sparklingVersion}
-                            docker tag sparkling-water-python:${sparklingVersion} $repoUrl/sw_kubernetes_repo/sparkling-water:python-${sparklingVersion}
-                            docker tag sparkling-water-scala:${sparklingVersion} $repoUrl/sw_kubernetes_repo/sparkling-water:scala-${sparklingVersion}
-                            docker push $repoUrl/sw_kubernetes_repo/sparkling-water:r-${sparklingVersion}
-                            docker push $repoUrl/sw_kubernetes_repo/sparkling-water:python-${sparklingVersion}
-                            docker push $repoUrl/sw_kubernetes_repo/sparkling-water:scala-${sparklingVersion}
-                            """
+                            withEnv(customEnv) {
+                                dir("./dist/build/zip/sparkling-water-${sparklingVersion}") {
+                                    sh """
+                                        ./bin/build-kubernetes-images.sh scala
```

Comment:
nit: three below sections seems to be very. Is it possible to generalize the logic into a function?

Reply:
Good idea, done!